tests/h/search/core_test.py

```diff
@@ -0,0 +1,243 @@
+
+        result = search.Search(pyramid_request, separate_replies=True).run({
+            "sort": "id", "order": "asc",
+        })
+
+        assert result.reply_ids == [reply_3.id, reply_2.id, reply_1.id]
+
+    def test_separate_replies_that_dont_match_the_search_are_still_included(self, factories,
+                                                                            pyramid_request,
+                                                                            Annotation):
+        """All replies to the matching annotations are included.
+
+        Even if the replies don't match the search query. As long as the
+        top-level annotation matches the search query, its replies will be
+        included in reply_ids.
+        """
+        user = factories.User()
+        reply_user = factories.User()
+        annotation = Annotation(userid=user.userid, shared=True)
+        reply = Annotation(userid=reply_user.userid, references=[annotation.id], shared=True)
+
+        result = search.Search(pyramid_request, separate_replies=True).run(
+            # The reply would not match this search query because it's from
+            # ``reply_user`` not ``user``.
+            params={"user": user.userid},
+        )
+
+        assert result.reply_ids == [reply.id]
+
+    def test_replies_from_different_pages_are_included(self, index, pyramid_request, Annotation):
+        """Replies that would not be on the same page are included."""
+        # First create an annotation and a reply.
+        annotation = Annotation(shared=True)
+        reply = Annotation(references=[annotation.id], shared=True)
+
+        # Update the annotation so that it would appear before the reply in the
+        # search results (if no separate_replies was given).
+        annotation.updated = datetime.datetime.now()
+        index(annotation)
+
+        # Now create 19 more annotations. Without separate_replies the
+        # annotation would be the 20th item in the search results (last item on
+        # the first page) and the reply would be pushed onto the second page.
+        for _ in range(19):
+            Annotation(shared=True)
+
+        result = search.Search(pyramid_request, separate_replies=True).run(params={"limit": 20})
+
+        # Even though the reply would have been on the second page of the
+        # search results, it is still included in reply_ids if
+        # separate_replies=True.
+        assert result.reply_ids == [reply.id]
+
+    def test_only_200_replies_are_included(self, pyramid_request, Annotation):
+        """No more than 200 replies can be included in reply_ids.
+
+        200 is the total maximum number of replies (to all annotations in
+        annotation_ids) that can be included in reply_ids.
+        """
+        annotation = Annotation(shared=True)
+        first_reply = Annotation(references=[annotation.id], shared=True)
+        for _ in range(200):
```

Comment:
Perhaps we should make the `200` value injectable and then use a smaller value to make the test run faster.

Reply:
Done